### ISSUE TITLE
Deprecation warnings

### ISSUE THREAD

**Issue by znicholls** [IAMconsortium/pyam#448]
At the moment, deprecation warnings are raised like https://github.com/IAMconsortium/pyam/blob/37e7c5eba23b3814dce5998efe0a09462a19ffcd/pyam/logging.py#L15

This means that they are a `UserWarning` rather than a `DeprecationWarning` i.e. they have the wrong type (which is annoying if you want to ignore them from a downstream package). I also don't think that logging them using a logger is the right approach (again, it should be on the downstream user to decide whether to log warnings or just leave them or whatever). I would suggest changing https://github.com/IAMconsortium/pyam/blob/37e7c5eba23b3814dce5998efe0a09462a19ffcd/pyam/logging.py#L15 to 

```python
def deprecation_warning(msg, type='This method'):
    """Issue deprecation warning"""
    warn = 'is deprecated and will be removed in future versions.'
    warnings.DeprecationWarning('{} {} {}'.format(type, warn, msg), stacklevel=3)  # stacklevel=3 so it's obvious what called the deprecated method/function in the first place
```

**Comment by znicholls** [comment]
Thoughts @gidden @danielhuppmann ?

**Comment by danielhuppmann** [comment]
@coroa already started a PR with a similar idea, see #444 - I'm all for this change as long as the warning is shown per default and isn't getting lost between stdout and stderr and nirvana...

**Comment by danielhuppmann** [comment]
closing as implemented via #444, please re-open if issues persist
